TESTS/network/wifi/main.cpp

```diff
@@ -0,0 +1,196 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "utest/utest.h"
+#include "unity/unity.h"
+#include "greentea-client/test_env.h"
+
+#include "mbed.h"
+
+using namespace utest::v1;
+
+/**
+ * WiFi tests require following macros to be defined:
+ * - WIFI_TEST_SSID - SSID of a network the test will try connecting to
+ * - WIFI_TEST_PASS - Passphrase that will be used to connecting to the network
+ * - WIFI_TEST_NETWORKS - List of network that presence will be asserted e.g. "net1", "net2", "net3"
+ */
+#if !defined(WIFI_TEST_SSID) || !defined(WIFI_TEST_PASS) || !defined(WIFI_TEST_NETWORKS)
+#error WIFI_TEST_NETWORKS, WIFI_TEST_PASS and WIFI_TEST_NETWORKS have to be defined for this test.
+#endif
+
+const char *networks[] = {WIFI_TEST_NETWORKS, NULL};
+
+/* We use singletons as some of the WiFi modules don't like to be initialized multiple times and using global objects
+   causes greentea serial to timeout.
+ */
+#if TARGET_UBLOX_EVK_ODIN_W2
+#include "OdinWiFiInterface.h"
+SingletonPtr<OdinWiFiInterface> wifi;
+#else
+#if !TARGET_FF_ARDUINO
+#error [NOT_SUPPORTED] Only Arduino form factor devices supported
```

Comment:
You make a good point, that wouldn't be the greatest. Could we at least have an option to provide pins that would avoid the unsupported error? It would be nice to be able to use this test with esp8266s on boards locally.

If it's too difficult, we can just modify the test directly if we need to test other boards.